outputparser/regex_parser.go

```diff
@@ -0,0 +1,73 @@
+package outputparser
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/tmc/langchaingo/schema"
+)
+
+// RegexParser is an output parser used to parse the output of an llm as a map.
+type RegexParser struct {
```

Comment:
Is the `Parser` suffix here acceptable? It is non-standard with the other existing, but I'm not a sure a struct named `Regex` is reasonable.